config/samples/sample-app-agent.yml

```diff
@@ -0,0 +1,132 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: quarkus-test-agent
+    app.kubernetes.io/component: quarkus-test-agent
+    app.kubernetes.io/instance: quarkus-test-agent
+  name: quarkus-test-agent
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      deployment: quarkus-test-agent
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        openshift.io/generated-by: OpenShiftNewApp
+      creationTimestamp: null
+      labels:
+        deployment: quarkus-test-agent
+    spec:
+      containers:
+      - env:
+        - name: CRYOSTAT_AGENT_APP_NAME
+          value: agent-test
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CRYOSTAT_AGENT_BASEURI
+          value: https://cryostat-sample.$(NAMESPACE).svc:8181
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: status.podIP
+        - name: CRYOSTAT_AGENT_CALLBACK
+          value: http://$(POD_IP):9977
+        - name: CRYOSTAT_AGENT_AUTHORIZATION
+          value: Bearer abcd1234
+        - name: KEYSTORE_PASS
+          valueFrom:
+            secretKeyRef:
+              key: KEYSTORE_PASS
+              name: cryostat-sample-keystore
+        - name: JAVA_OPTS_APPEND
+          value: |-
+            -Djavax.net.ssl.trustStore=/var/run/secrets/myapp/truststore.p12
+            -Djavax.net.ssl.trustStorePassword=$(KEYSTORE_PASS)
+        image: quay.io/andrewazores/quarkus-test@sha256:96f0f8bf0878a0843df08f9fec8ee9cf9c0ea09ee8c462b2c28ae504085a8b1b
+        imagePullPolicy: IfNotPresent
+        name: quarkus-test-agent
+        ports:
+        - containerPort: 10010
+          protocol: TCP
+        - containerPort: 8080
+          protocol: TCP
+        - containerPort: 8443
+          protocol: TCP
+        - containerPort: 9097
+          protocol: TCP
+        resources: {}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /var/run/secrets/myapp/truststore.p12
+          name: truststore
+          subPath: truststore.p12
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext:
+        runAsNonRoot: true
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - name: truststore
+        secret:
+          defaultMode: 420
```

Comment:
Some reasons to set this permission bit to `420`?

Reply:
I think I got this from an example that @ebaron had put together elsewhere, but I don't know where that came from originally. I can set up a testing environment again and experiment with leaving this configuration out and see if that works.

Reply:
I think `420` should already be the default value in decimal (octal `0644`).

```bash
oc explain pods.spec.volumes.secret.defaultMode
KIND:     Pod
VERSION:  v1

FIELD:    defaultMode <integer>

DESCRIPTION:
     defaultMode is Optional: mode bits used to set permissions on created files
     by default. Must be an octal value between 0000 and 0777 or a decimal value
     between 0 and 511. YAML accepts both octal and decimal values, JSON
     requires decimal values for mode bits. Defaults to 0644. Directories within
     the path are not affected by this setting. This might be in conflict with
     other options that affect the file mode, like fsGroup, and the result can
     be other mode bits set.
```